### DISCUSSION TITLE
Filebeat command to build logstash pipeline

### DISCUSSION THREAD

**Discussion by gobrien1231** [Security-Onion-Solutions/securityonion#6244] [Unsupported Versions] [answered]
Regarding ingesting netflow using the guide "Ingesting Netflow in Security Onion" by Matthew Gracie on the SO youtube channel... 

after running through the video guide, and running this command: 
# docker exec -i so-filebeat filebeat setup modules -pipelines -modules netflow -c /usr/share/filebeat/module-setup.yml

i see this configuration file is still empty on my system: /usr/share/filebeat/module-setup.yml 

So i'm wondering if that may be part of my issue on getting the netflow.log event dataset to appear in my hunt interface. Any thoughts?

**Comment by InfosecGoon** [comment, marked as answer]
Good morning!

Just a few questions to try and narrow down the problem:

1. What version of SO are you using? The Filebeat module functionality from the video was introduced in 2.3.60, so if you're on an older version it's not going to work properly.

2. When you ran that command, did you get any response from the CLI? It should say "Loaded Ingest pipelines".

3. To verify whether the pipelines are there, try running a "tail -f" on the Logstash log file at /opt/so/logs/logstash/logstash.log. If the Netflow logs are coming into the system and then being dropped because it doesn't know what pipeline to use, there will be errors there.

--Matt

**Reply by gobrien1231** [reply]
Thanks for the reply! I'm running 2.3.80. I did indeed get the "loaded Ingest Pipleline" feedback from the cli. I'm not seeing anything in the logstash.log pertaining to netflow. 
So it seems I've got something else going on in my system...or I missed something during the config.  I'll keep digging and holler when i've got some more info to share on this.

**Reply by gobrien1231** [reply]
Hey Matt - so i figure this out. My issue was with permissions on the local filebeat init.sls file. Once i properly set the ownership of the file to my user, netflow is coming through now. Its been really handy for correlations so far. Thanks again for the video.

**Reply by tomjones1977** [reply]
@gobrien1231 what permissions/ownership do you have on your filebeat init.sls file?  I'm trying to achieve the same goal and having the same issue.

When running filebeat in debug mode I can see it ingests and parses the netflow data correctly, but then the data disappears.  I can't see any errors in the filebeat or logstash logfiles that would suggest a problem, but the data just vanishes and never makes it to the ES database.

Edit: It turns out that you need to run the docker command on the search nodes as well as the master in a distributed setup.  The Logstash logs on the search node were going hell for leather showing ingest failures due to missing pipelines.  Adding these on the search nodes fixed the issue.

**Reply by gobrien1231** [reply]
@tomjones1977 I had just missed assigning ownership of that file, initially. after I took ownership of the init file with my user, it started to work.
